### PR TITLE
implemented EXPLAIN for select statements

### PR DESCRIPTION
**Pull request by dobe**

### CODE REVIEW

File: sql/src/main/java/io/crate/action/sql/TransportBaseSQLAction.java [hunk 1]
```diff
@@ -296,12 +298,19 @@ public void onFailure(Throwable e) {
...
-            PlanPrinter printer = new PlanPrinter();
-            logger.trace(printer.print(plan));
+            String json = null;
+            try {
+                json = JsonXContent.contentBuilder()
+                        .humanReadable(true)
+                        .prettyPrint()
+                        .value(PlanPrinter.objectMap(plan)).bytes().toUtf8();
+            } catch (IOException e) {
+                e.printStackTrace();
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/TransportBaseSQLAction.java]: don't just print it.. use logger or raise as runtime exception

File: sql/src/main/java/io/crate/analyze/ExplainAnalyzedStatement.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    public <C, R> R accept(AnalyzedStatementVisitor<C, R> analyzedStatementVisitor, C context) {
+        return analyzedStatementVisitor.visitExplainStatement(this, context);
+    }
+
+    public AnalyzedStatement statement() {
+        return statement;
+    }
+
+    @Override
+
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/ExplainAnalyzedStatement.java]: remove newline between override

File: sql/src/test/java/io/crate/planner/PlannerTest.java [hunk 3]
```diff
@@ -2088,4 +2100,29 @@ public void testNonDistributedGroupByProjectionHasShardLevelGranularity() throws
...
+            ExplainPlan plan = plan("explain " + statement);
+            assertNotNull(plan);
+            assertNotNull(plan.subPlan());
+        }
+    }
+
+    @Test
+    public void testPrinter() throws Exception {
+        for (String statement : EXPLAIN_TEST_STATEMENTS) {
+            System.out.println(statement);
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/planner/PlannerTest.java]: remove println

**Comment by seut** [comment]
lgtm

**Comment by dobe** [comment]
already merged, somehow gh got mixed up

**Pull request closed without merging**